src/options.c

```diff
@@ -369,131 +415,80 @@ JOHN_USAGE_REGEX \
-	puts("--show=formats            show information about hashes in a file (JSON)");
-	puts("--show=invalid            show lines that are not valid for selected format(s)");
-	puts("--skip-self-tests         skip self tests");
-	puts("--test-full=LEVEL         run more thorough self-tests");
-	puts("--stress-test[=TIME]      loop self tests forever");
-	puts("--no-mask                 used with --test for alternate benchmark w/o mask");
-#ifdef HAVE_FUZZ
-	puts("--fuzz[=DICTFILE]         fuzz formats' prepare(), valid() and split()");
-	puts("--fuzz-dump[=FROM,TO]     dump the fuzzed hashes between FROM and TO to file pwfile.format");
-#endif
-	puts("--input-encoding=NAME     input encoding (alias for --encoding)");
-	puts("--internal-codepage=NAME  codepage used in rules/masks (see doc/ENCODINGS)");
-	puts("--target-encoding=NAME    output encoding (used by format, see doc/ENCODINGS)");
-	puts("--tune=HOW                tuning options (auto/report/N)");
-	puts("--incremental-charcount=N override CharCount for incremental mode");
-	puts("--subsets-required=N      The N first characters of \"subsets\" charset are");
-	puts("                          the \"required set\" (see doc/SUBSETS)");
-	puts("--subsets-min-diff=N      Minimum unique characters in subset");
-	puts("--subsets-max-diff=[-]N   Maximum unique characters in subset (negative N is");
-	puts("                          relative to word length)");
-#ifdef HAVE_OPENCL
-	puts("\nOpenCL options:");
-	puts("--force-scalar            force scalar mode");
-	puts("--force-vector-width=N    force vector width N");
-	puts("--lws=N                   force local worksize N");
-	puts("--gws=N                   force global worksize N");
-#endif
-#if HAVE_LIBGMP || HAVE_INT128 || HAVE___INT128 || HAVE___INT128_T
-	puts("\nPRINCE mode options:");
-	puts("--prince-loopback[=FILE]  fetch words from a .pot file");
-	puts("--prince-elem-cnt-min=N   minimum number of elements per chain (1)");
-	puts("--prince-elem-cnt-max=[N] maximum number of elements per chain (negative N is");
-	puts("                          relative to word length) (8)");
-	puts("--prince-skip=N           initial skip");
-	puts("--prince-limit=N          limit number of candidates generated");
-	puts("--prince-wl-dist-len      calculate length distribution from wordlist");
-	puts("--prince-wl-max=N         load only N words from input wordlist");
-	puts("--prince-case-permute     permute case of first letter");
-	puts("--prince-mmap             memory-map infile (not available with case permute)");
-	puts("--prince-keyspace         just show total keyspace that would be produced");
-	puts("                          (disregarding skip and limit)");
-#endif
-	puts("");
-}
-
-void opt_init(char *name, int argc, char **argv, int show_usage)
-{
-	if (show_usage)
-		print_usage(name);
+			printf("%s", JOHN_USAGE_FORMAT);
+		}
+	} else if(argc < 2) {
+		if (john_main_process)
+			printf(JOHN_BANNER, name);
+		exit(0);
+	}
```

Comment:
Still one `exit()` missing and that is why you suddenly got the 'Unknown option: "--help"'.

You need one `exit(0);` after line 486 here, that is INSIDE the `if (argc == 2 && ...)` block but OUTSIDE the `if (john_main_process)` block.

Reply:
BTW while at it, there's a space missing:
```diff
- 	} else if(argc < 2) {
+ 	} else if (argc < 2) {
```